### DISCUSSION TITLE
How can I user the rest api to get a named repo for the authenticated user?

### DISCUSSION THREAD

**Discussion by Mufasa** [octokit/octokit.js#2483] [General] [answered]
I have a non-organisational GitHub repo. I have worked out how to get a list of all the repos for the authenticated user as follows:
```
import { Octokit } from 'octokit';

const octokit = new Octokit({ auth: 'MY_GITHUB_PERSONAL_ACCESS_TOKEN' });

const doit = async (): Promise<void> => {
  const response = await octokit.rest.repos.listForAuthenticatedUser({ });

  console.log(JSON.stringify(response, null, 2));
};

void doit();
```
However, I am struggling to work out how to get a specific named repo using this library. I do not wish to list all the repos and then search through them to find the one I need. I want to be able to directly get the specific named repo.

Is there a way of achieving this?

Also, is there a way to get the commit details given its commit SHA (again for a non-organisational repo)?

**Comment by gr2m** [comment, marked as answer]
```js
const { data: repository}  = await octokit.rest.repos.get({owner: "octokit", repo: "octokit.js"})
```

If you want to get the current user's username, you can do

```js
const { data: user } = await octokit.rest.users.getAuthenticated()
// user.login
```

> Also, is there a way to get the commit details given its commit SHA (again for a non-organisational repo)?

It's the same for user and org repositories

```js
const { data: commit } = await octokit.rest.repos.getCommit({ owner: "octokit", repo: "octokit.js", ref: "46b5f2fd41ef6d2ac23653ca302d6ba7e5559c02" })
```

**Reply by Mufasa** [reply]
The above only works for an organisational repo. I have a user repo - how do I get those details for a user repo?

e.g. for an organisational repo, you can list the repos in it by running `GET https://api.github.com/orgs/OWNER/repos` where as for a user repo you would need to run `GET https://api.github.com/users/USER_NAME/repos`.

`octokit.rest.repos.get` seems to be doing an organisational repo lookup unless I have misunderstood something here?

**Reply by gr2m** [reply]
> `octokit.rest.repos.get` seems to be doing an organisational repo lookup unless I have misunderstood something here?

no, it works the same for both types of repository owners. The `owner` parameter can be set to either a user login or an organization login

**Reply by Mufasa** [reply]
Ok, I think I may know why its not working for me now. When I run `const response = await octokit.rest.repos.listForAuthenticatedUser();` it only seems to return a list of the **public** repos. How can I get the **private** ones as well?

**Reply by Mufasa** [reply]
@gr2m Apologies - I finally figured it out. My personal access token only included these scopes:
- repo:statusAccess
- repo_deploymentAccess
- public_repo

After I changed it to include all the **repo** scopes then I was able to get the info I wanted. Sorry for my ignorance.

**Reply by gr2m** [reply]
no worries
